### PR TITLE
Fix parsing of album and views in parse_playlist_items

### PR DESCRIPTION
**Pull request by fischcode**
Parse album and views columns as a function of flex column count and is_album property to cater for varying layouts with three and four flex columns

### CODE REVIEW

**Comment by fischcode** [comment]
@sigma67 Please note that the three commits are independent contributions an that there are no dependencies between them whatsoever

File: ytmusicapi/mixins/playlists.py [hunk 1]
```diff
@@ -112,10 +112,12 @@ def get_playlist(
         if not own_playlist:
             header = response["header"]["musicDetailHeaderRenderer"]
             playlist["privacy"] = "PUBLIC"
+            playlist["own_playlist"] = False
```
Comment (sigma67) [inline ytmusicapi/mixins/playlists.py:115]: I would prefer to call this `owned`, much simpler.

Also you can just set the attribute outside the if and reuse the local variable `own_playlist`, much simpler

File: ytmusicapi/parsers/playlists.py [hunk 2]
```diff
@@ -44,15 +44,13 @@ def parse_playlist_items(results, menu_entries: Optional[List[List]] = None, is_
...
 
-        album = parse_song_album(data, 2)
+        album = parse_song_album(data, flex_column_count-1) if not is_album else None
 
-        views = None
-        if album and album["id"] is None:
-            # views currently only present on albums and formatting is localization-dependent -> no parsing
-            if (views := (get_item_text(data, 2))) is not None:
-                album = None
+        views = get_item_text(data, 2) if flex_column_count == 4 or is_album else None
```
Comment (sigma67) [inline ytmusicapi/parsers/playlists.py:53]: > flex_column_count == 4 or is_album

which cases does this condition cover? `is_album` is obvious, but when does the `count==4` apply when it's not an album?

File: ytmusicapi/parsers/podcasts.py [hunk 3]
```diff
@@ -103,6 +103,7 @@ def parse_episodes(results) -> List[Dict]:
     episodes = []
     for result in results:
         data = nav(result, ["musicMultiRowListItemRenderer"])
+        thumbnails = nav(data, THUMBNAILS)[:2]
```
Comment (sigma67) [inline ytmusicapi/parsers/podcasts.py:106]: why are you only taking the first two items? just leave out the `[:2]`

**Review by sigma67 — COMMENTED** [review]
Thanks for the contribution, looks good mostly. Some minor comments & please fix linter issues :)

File: ytmusicapi/parsers/playlists.py (hunk 2 above)
Comment (fischcode) [reply ytmusicapi/parsers/playlists.py:53]: I am covering
- Album with three columns, the last of which is views/plays:
grafik
- Playlist with three columns, the last of which is album:
grafik
- Artist songs from Artist overview with four columns, the third of which is views/plays: and the fourth of which is album:
grafik
- Artist song details with four columns, the third of which is views/plays: and the fourth of which is album:
grafik

File: ytmusicapi/parsers/podcasts.py (hunk 3 above)
Comment (fischcode) [reply ytmusicapi/parsers/podcasts.py]: Some podcasts episodes appear to contain technical images in addition to the episode related images, e.g.
https://i.ytimg.com/vi/dPw_SXs1QOA/hq720.jpg:
Podcast 'Smosh Dungeons and Dragons', beginning of first episode:
`podcast = [{
        'musicMultiRowListItemRenderer': {
            'trackingParams': 'CHEQ3-AIGAAiEwjssYvPvceEAxWU0UIFHRKZBZk=',
            'thumbnail': {
                'musicThumbnailRenderer': {
                    'thumbnail': {
                        'thumbnails': [{
                                'url': 'https://i.ytimg.com/vi/f-vhvYvI0ok/hqdefault.jpg?sqp=-oaymwEWCOADEI4CIAQqCggAEOADGC0guwJIWg&rs=AMzJL3lYQ_uibdN_dDCw6PWruHiO_dolmQ',
                                'width': 480,
                                'height': 270
                            }, {
                                'url': 'https://i.ytimg.com/vi/f-vhvYvI0ok/hq720.jpg?sqp=-oaymwEKCKAGEMIDIABIWg&rs=AMzJL3k6ozV_-vU-izDtPVWu-lbKy5wgAw',
                                'width': 800,
                                'height': 450
                            }, {
                                'url': 'https://i.ytimg.com/vi/f-vhvYvI0ok/hq720.jpg',
                                'width': 1280,
                                'height': 720
                            }
                        ]
                    },
                    'thumbnailCrop': 'MUSIC_THUMBNAIL_CROP_UNSPECIFIED',
                    'thumbnailScale': 'MUSIC_THUMBNAIL_SCALE_ASPECT_FIT',
                    'trackingParams': 'CHsQhL8CIhMI7LGLz73HhAMVlNFCBR0SmQWZ'
                }
            },
`
Usually I display the last image in the list as it normally has the best resolution. But here that would result in the wrong image being displayed. So far, I haven't found episodes where such a technical image appeared before third position. Not having come up with a better idea so far, I therefore had decided to drop all thumbnails after position two.
Just now I realize that the third image's URL doesn't contain a question mark, so an alternative approach would be to filter for thumbnails with URLs containing one:
thumbnails = list(filter(lambda tn: '?' in tn['url'], nav(data, THUMBNAILS)))

File: ytmusicapi/mixins/playlists.py (hunk 1 above)
Comment (fischcode) [reply ytmusicapi/mixins/playlists.py]: OK, changed

File: ytmusicapi/parsers/playlists.py (hunk 2 above)
Comment (sigma67) [reply ytmusicapi/parsers/playlists.py:53]: Thanks for this overview, that explains it!

File: ytmusicapi/parsers/podcasts.py (hunk 3 above)
Comment (sigma67) [reply ytmusicapi/parsers/podcasts.py]: I realize that the same image is present in multiple resolutions. This is the case for other thumbnails as well.

`ytmusicapi` always returns all thumbnails - it's up to the user to choose the appropriate size. You never know the precise use case in advance, so can't be too judgy on the input data filtering.

**Comment by fischcode** [comment]
Sorry, I don't understand: what is linter?

**Review by sigma67 — APPROVED** [review]

**Pull request merged**